app/lib/features/chat/pages/room_profile_page.dart

```diff
@@ -361,32 +339,55 @@ class _RoomProfilePageState extends ConsumerState<RoomProfilePage> {
             child: Text(L10n.of(context).no),
           ),
           ElevatedButton(
-            onPressed: () async {
-              Navigator.of(context, rootNavigator: true).pop();
-              EasyLoading.show(status: L10n.of(context).leavingRoom);
-              var res = await _handleLeaveRoom(ref, widget.roomId);
-              if (!mounted) return;
-              if (res) {
-                if (context.mounted) {
-                  EasyLoading.dismiss();
-                  context.goNamed(Routes.chat.name);
-                }
-              } else {
-                EasyLoading.dismiss();
-                EasyLoading.showError(
-                  L10n.of(context).someErrorOccurredLeavingRoom,
-                );
-              }
-            },
+            onPressed: _handleLeaveRoom,
             child: Text(L10n.of(context).yes),
           ),
         ],
       ),
     );
   }
 
-  Future<bool> _handleLeaveRoom(WidgetRef ref, String roomId) async {
-    final convo = await ref.read(chatProvider(roomId).future);
-    return await convo.leave();
+  Future<void> _handleLeaveRoom() async {
+    Navigator.of(context, rootNavigator: true).pop();
+    EasyLoading.show(status: L10n.of(context).leavingRoom);
+    final convo = await ref.read(chatProvider(widget.roomId).future);
+    final res = await convo.leave();
+    if (!context.mounted) {
+      EasyLoading.dismiss();
+      return;
+    }
+    if (res) {
+      EasyLoading.dismiss();
+      context.goNamed(Routes.chat.name);
+    } else {
+      EasyLoading.showError(
+        L10n.of(context).someErrorOccurredLeavingRoom,
+        duration: const Duration(seconds: 3),
+      );
+    }
+  }
+
+  void _handleInvite(Member membership) {
+    if (membership.canString('CanInvite') == true) {
+      context.pushNamed(
+        Routes.spaceInvite.name,
+        pathParameters: {'spaceId': widget.roomId},
+      );
+    } else {
+      EasyLoading.showError(
+        L10n.of(context).notEnoughPowerLevelForInvites,
+        duration: const Duration(seconds: 3),
+      );
+    }
+  }
+
+  Future<void> _handleShare() async {
+    final convo = await ref.read(chatProvider(widget.roomId).future);
+    final roomLink = await convo.permalink();
+    if (!mounted) return;
+    Share.share(
+      roomLink,
+      subject: L10n.of(context).linkToChat,
+    );
```

Comment:
this entire thing should be in a try-catch with log and showError

Reply:
Fixed by https://github.com/acterglobal/a3/pull/1580/commits/ee3b10a880117604995042e97df70be9cd4ad768